cocotb/share/include/exports.h

```diff
@@ -0,0 +1,16 @@
+// Copyright cocotb contributors
+// Licensed under the Revised BSD License, see LICENSE for details.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef COCOTB_EXPORTS_H_
+#define COCOTB_EXPORTS_H_
+
+#if defined(__linux__) || defined(__APPLE__)
```

Comment:
It would be good to include a comment above this line saying:

* Why we care about visibility / dll settings.
* How the macros in this file are intended to be used.